docs/service-overview.md

```diff
@@ -0,0 +1,947 @@
+
+    siteA("Group: SiteA
+    Group ID: group-ad636092-fbc4-446c-ab2f-9df16719613a
+    Switch Serials = [GACXXXXXX]
+    Users: [{username = userconsulting, role = admin}]")
+
+    siteB("
+    Group: SiteB
+    Group ID: group-79ef6ff1-98d1-4fa7-98f3-7d98af4ee6ee
+    Users: [{username = siteb, role = requestor}]")
+
+    orgBlock --> topGroup
+    topGroup --> siteA
+    topGroup --> siteB
+```
+
+### Authorization
+
+The above figure shows the relationships between different entities in the
+system. This is the organization tree. The root of the tree is the organization
+itself. In this example, the organization has one child group, which has two
+child groups. Each group has a set of pinned domain certs, a set of users, and a
+set of serials assigned to it. This system has 3 roles: **ADMIN**, **ASSIGNER**,
+and **REQUESTOR**. An **ADMIN** can do everything that an **ASSIGNER** can do.
+An **ASSIGNER** can do everything a **REQUESTOR** can do.
+
+Each user has access to the group that it belongs to along with all of its
+children, i.e., the user has access to the subtree rooted at the group that the
+user belongs to. One user can belong to multiple groups and access multiple
+subtrees in the hierarchy. The user has the same permission across the subtree
+that it has been assigned at the root group of the subtree. If two subtrees
+overlap for a user, the permission that gives the user higher access is applied.
+
+In the example shown in the figure, the user with `username=siteb` cannot request
+an ownership voucher for the serial JGEXXXXXX because the serial belongs to the
+parent group. The user with username=useracm can request an ownership voucher
+for the serial with serial number GACXXXXXX because the user has access to the
+parent group of the serial.
+
+## Serial Numbers
+
+Ownership vouchers can be issued for network elements as identified by their
+serial number. Some products may also have a TPM Public Key associated with
+them. By default, it is assumed that all the customer-owned serials are added to
+the root group. Users with the **ASSIGNER** or **ADMIN** role can move/add
+serials within the tree from there. A serial can be assigned to a single group
+apart from the root group. See the `/GetSerial` RPC for an example of the
+details available for a given serial number.
+
+## Pinned Domain Certificates (PDCs)
+
+Pinned Domain Certificates (aka PDCs) are the roots of trust used in the
+ownership voucher, along with “revocation checks” and an “expiry time”. A
+certificate ID is returned when a pinned domain cert is added to a group. While
+creating ownership vouchers, this cert ID is used to reference the certificate.
+See the `/GetDomainCert` RPC for an example of this data.
+
+## Roles
+
+Users are assigned to **ADMIN** or **REQUSTER** roles. An **ADMIN** can do
+everything that a (**ASSIGNER**, **REQUESTOR**) can do, along with requesting
+Ownership Vouchers. See `/GetGroup` RPC for an example. It is a subfield of
+group data.
```

Comment:
I think it would be useful to mention that roles are always assigned within the context of the group. Perhaps that's what 'It is a subfield of group data.' means?